src/method/method_query_handler.cpp

```diff
@@ -235,6 +235,217 @@ namespace cubmethod
     return info;
   }
 
+  get_generated_keys_info
+  query_handler::generated_keys ()
+  {
+    int error = NO_ERROR;
+
+    get_generated_keys_info info;
+
+    int stmt_type;
+    DB_QUERY_RESULT *qres = (DB_QUERY_RESULT *) m_query_result.result;
+    if (qres == NULL)
+      {
+	// TODO: proper error code
+	m_error_ctx.set_error (METHOD_CALLBACK_ER_INTERNAL, NULL, __FILE__, __LINE__);
+	error = ER_FAILED;
+      }
+    else
+      {
+	if (qres->type == T_SELECT)
+	  {
+	    stmt_type = qres->res.s.stmt_type;
+	    error = get_generated_keys_server_insert (info, *qres);
+	  }
+	else if (qres->type == T_CALL)
+	  {
+	    stmt_type = m_query_result.stmt_type;
+	    error = get_generated_keys_client_insert (info, *qres);
+	  }
+	else
+	  {
+	    // TODO: proper error code
+	    m_error_ctx.set_error (METHOD_CALLBACK_ER_INTERNAL, NULL, __FILE__, __LINE__);
+	    error = ER_FAILED;
+	  }
+      }
+
+    if (error == NO_ERROR)
+      {
+	/* set qresult_info */
+	query_result_info &result_info = info.qresult_info;
+	result_info.stmt_type = stmt_type;
+	result_info.tuple_count = info.generated_keys.tuples.size ();
+	result_info.query_id = -1; /* initialized value, intead of garbage */
+      }
+
+    return info;
```

Comment:
error is set but not used.

Reply:
if local variable, `error` is set NO_ERROR, safely able to set qresult_info (line 273 ~ 280). Errors are handled by m_error_ctx and passed to callback_handler.